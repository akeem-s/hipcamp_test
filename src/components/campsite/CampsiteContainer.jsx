import React, { Component } from 'react';
// components
import Feature from './Feature';
import { FEATURES } from '../../assets/camp_features';

export default class CampsiteContainer extends Component {
  constructor(props){
    super(props);
    this.state = {
      features:[],
    };
  }

  componentDidMount(){
    // Note: I am setting state here because I am treating this data like an api response
    // and the React docs suggest instantiating API calls in componentDidMount. Setting state
    // here can have performance implications, which I would consider if dealing with real
    // data. The state of this container would normally manage the state of the
    // api call (isLoading, hasError, etc)
    this.setState({
      features: FEATURES,
    });
  }

  render(){
    return (
      <div className='campsiteContainer'>
        <h2 className='campsiteName'>Campsite Name Placeholder</h2>
        {this.state.features.map((f)=>{ return <Feature key={f.title} title={f.title} presence={f.presence} subfeatures={f.subfeatures}/>; })}
      </div>
    );
  }
}
