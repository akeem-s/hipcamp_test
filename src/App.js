import React, { Component } from 'react';
import './styles.scss'
// Components
import CampsiteContainer from './components/campsite/CampsiteContainer';
import Header from './shared/Header';

class App extends Component {
  render() {
    return (
      <div>
        <Header />
        <div className='mainContentContainer'>
          <CampsiteContainer/>
        </div>
      </div>
    );
  }
}

export default App;
