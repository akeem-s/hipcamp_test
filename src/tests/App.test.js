import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import CampsiteContainer from '../components/campsite/CampsiteContainer';

describe(App.name, () => {
  let wrapper;
  const getWrapper = () => {
    const component = <App />;
    return shallow(component);
  };

  beforeEach(() => {
    wrapper = getWrapper();
  });

  test('component renders', () => {
    expect(wrapper.length).toEqual(1);
  });

  test('includes CampsiteContainer', () => {
    expect(wrapper.find(CampsiteContainer).length).toEqual(1);
  });

  test('snapshot should contain expected DOM elements', () => {
    const renderedValue = renderer.create(wrapper).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
