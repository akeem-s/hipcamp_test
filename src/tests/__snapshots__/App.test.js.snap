// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should contain expected DOM elements 1`] = `
<div>
  <div
    className="header"
  >
    <div
      className="headerLogoContainer"
    >
      <img
        alt="hipcampLogo"
        className="headerLogo"
        src="hipcampNewLogo.png"
      />
    </div>
  </div>
  <div
    className="mainContentContainer"
  >
    <div
      className="campsiteContainer"
    >
      <h2
        className="campsiteName"
      >
        Campsite Name Placeholder
      </h2>
      <div
        className="feature null"
        onClick={[Function]}
      >
        <div
          className="featureDetailsContainer"
        >
          <p
            className="featureTitle"
          >
            Toilet
          </p>
          <span
            aria-hidden={true}
            className="fa fa-check customFaCheck customFa"
          />
        </div>
      </div>
      <div
        className="feature null"
        onClick={[Function]}
      >
        <div
          className="featureDetailsContainer"
        >
          <p
            className="featureTitle"
          >
            Pets allowed
          </p>
          <span
            aria-hidden={true}
            className="fa fa-check customFaCheck customFa"
          />
        </div>
      </div>
      <div
        className="feature subfeaturesPresent"
        onClick={[Function]}
      >
        <div
          className="featureDetailsContainer"
        >
          <p
            className="featureTitle"
          >
            Shower
          </p>
          <span
            aria-hidden={true}
            className="fa fa-ban customFaBan customFa"
          />
        </div>
      </div>
      <div
        className="feature subfeaturesPresent"
        onClick={[Function]}
      >
        <div
          className="featureDetailsContainer"
        >
          <p
            className="featureTitle"
          >
            Trash
          </p>
          <span
            aria-hidden={true}
            className="fa fa-check customFaCheck customFa"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
